### ISSUE TITLE
[Bug] Latest Fabric API crashes on startup with BCLib

### ISSUE THREAD

**Issue by aisukuma** [quiqueck/BCLib#4]
### What happened?

Fabric API pushed an update today which causes Minecraft to not boot up with BCLib enabled.

Downgrading to Fabric API 0.55.3+1.19 works for the time being.

### BCLib

2.0.4

### Fabric API

0.56.0+1.19

### Fabric Loader

0.14.7

### Minecraft

1.19

### Relevant log output

```shell
[23:41:55] [main/INFO]: Loading Minecraft 1.19 with Fabric Loader 0.14.7
[23:41:55] [main/INFO]: Loading 51 mods:
	- bclib 2.0.4
	- fabric 0.56.0+1.19
	- fabric-api-base 0.4.9+e62f51a3a9 via fabric
	- fabric-api-lookup-api-v1 1.6.7+9ff28f40a9 via fabric
	- fabric-biome-api-v1 9.0.13+b2a4a624a9 via fabric
	- fabric-blockrenderlayer-v1 1.1.18+9ff28f40a9 via fabric
	- fabric-command-api-v1 1.2.5+f71b366fa9 via fabric
	- fabric-command-api-v2 2.1.1+cda9839ba9 via fabric
	- fabric-commands-v0 0.2.22+df3654b3a9 via fabric
	- fabric-containers-v0 0.1.27+df3654b3a9 via fabric
	- fabric-content-registries-v0 3.1.0+4a3ec384a9 via fabric
	- fabric-convention-tags-v1 1.0.8+37622d24a9 via fabric
	- fabric-crash-report-info-v1 0.2.3+bd0a0d4aa9 via fabric
	- fabric-data-generation-api-v1 5.0.1+2a5b9210a9 via fabric
	- fabric-dimensions-v1 2.1.27+72da3b3da9 via fabric
	- fabric-entity-events-v1 1.4.16+9ff28f40a9 via fabric
	- fabric-events-interaction-v0 0.4.26+9ff28f40a9 via fabric
	- fabric-events-lifecycle-v0 0.2.25+df3654b3a9 via fabric
	- fabric-game-rule-api-v1 1.0.19+18990361a9 via fabric
	- fabric-item-api-v1 1.5.4+767db26ea9 via fabric
	- fabric-item-groups-v0 0.3.23+9ff28f40a9 via fabric
	- fabric-key-binding-api-v1 1.0.18+9ff28f40a9 via fabric
	- fabric-keybindings-v0 0.2.16+df3654b3a9 via fabric
	- fabric-lifecycle-events-v1 2.1.0+33fbc738a9 via fabric
	- fabric-loot-api-v2 1.0.1+9e7660c6a9 via fabric
	- fabric-loot-tables-v1 1.1.1+9e7660c6a9 via fabric
	- fabric-message-api-v1 1.0.1+513f4a59a9 via fabric
	- fabric-mining-level-api-v1 2.1.8+33fbc738a9 via fabric
	- fabric-models-v0 0.3.15+9ff28f40a9 via fabric
	- fabric-networking-api-v1 1.0.27+7fe97409a9 via fabric
	- fabric-networking-v0 0.3.14+df3654b3a9 via fabric
	- fabric-object-builder-api-v1 4.0.5+9ff28f40a9 via fabric
	- fabric-particles-v1 1.0.8+dc39553aa9 via fabric
	- fabric-registry-sync-v0 0.9.16+92cf9a3ea9 via fabric
	- fabric-renderer-api-v1 1.0.8+9ff28f40a9 via fabric
	- fabric-renderer-indigo 0.6.6+9ff28f40a9 via fabric
	- fabric-renderer-registries-v1 3.2.18+df3654b3a9 via fabric
	- fabric-rendering-data-attachment-v1 0.3.12+9ff28f40a9 via fabric
	- fabric-rendering-fluids-v1 3.0.5+9ff28f40a9 via fabric
	- fabric-rendering-v0 1.1.20+df3654b3a9 via fabric
	- fabric-rendering-v1 1.10.14+9ff28f40a9 via fabric
	- fabric-resource-conditions-api-v1 2.0.9+e62f51a3a9 via fabric
	- fabric-resource-loader-v0 0.5.3+9e7660c6a9 via fabric
	- fabric-screen-api-v1 1.0.23+9ff28f40a9 via fabric
	- fabric-screen-handler-api-v1 1.2.6+9ff28f40a9 via fabric
	- fabric-textures-v0 1.0.18+9ff28f40a9 via fabric
	- fabric-transfer-api-v1 2.0.9+e62f51a3a9 via fabric
	- fabric-transitive-access-wideners-v1 1.1.1+9e7660c6a9 via fabric
	- fabricloader 0.14.7
	- java 17
	- minecraft 1.19
[23:41:56] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/C:/Users/vinni/curseforge/minecraft/Install/libraries/net/fabricmc/sponge-mixin/0.11.4+mixin.0.8.5/sponge-mixin-0.11.4+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT
[23:41:56] [main/INFO]: Compatibility level set to JAVA_17
[23:41:57] [main/WARN]: Configuration conflict: there is more than one oshi.properties file on the classpath
[23:41:57] [main/WARN]: Configuration conflict: there is more than one oshi.architecture.properties file on the classpath
[23:42:00] [main/INFO]: Building optimized datafixer
[23:42:03] [main/INFO]: [bclib] Added Config bclib.main to auto sync (content diff)
[23:42:03] [main/INFO]: [bclib] Added Config bclib.recipes to auto sync (file hash)
[23:42:03] [main/INFO]: [bclib] Make sure features are loaded from terrablender for BCLib - Nether BiomeSource (3aeb6a57, version=18, seed=0, biomes=5)
[23:42:03] [main/ERROR]: Minecraft has crashed!
net.fabricmc.loader.impl.FormattedException: java.lang.ExceptionInInitializerError
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:463) ~[fabric-loader-0.14.7.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.7.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.7.jar:?]
Caused by: java.lang.ExceptionInInitializerError
	at net.minecraft.class_2378.<clinit>(class_2378.java:326) ~[client-intermediary.jar:?]
	at net.minecraft.class_2966.method_12851(class_2966.java:50) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:154) ~[fabric-loader-0.14.7-1.19.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.7.jar:?]
	... 2 more
Caused by: java.lang.ClassCastException: class net.minecraft.class_5321 cannot be cast to class net.minecraft.class_6880 (net.minecraft.class_5321 and net.minecraft.class_6880 are in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @21a947fe)
	at net.fabricmc.fabric.impl.biome.TheEndBiomeData$RegistryKeyHashStrategy.equals(TheEndBiomeData.java:173) ~[fabric-biome-api-v1-9.0.13+b2a4a624a9-bc490388a6b05b0c.jar:?]
	at it.unimi.dsi.fastutil.objects.Object2ObjectOpenCustomHashMap.containsKey(Object2ObjectOpenCustomHashMap.java:399) ~[fastutil-8.5.6.jar:?]
	at net.fabricmc.fabric.impl.biome.TheEndBiomeData$Overrides.bcl_canGenerateAsEndBarrensBiome(TheEndBiomeData.java:543) ~[fabric-biome-api-v1-9.0.13+b2a4a624a9-bc490388a6b05b0c.jar:?]
	at org.betterx.bclib.interfaces.TheEndBiomeDataAccessor.bcl_canGenerateInEnd(TheEndBiomeDataAccessor.java:18) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.api.v2.generator.BCLibEndBiomeSource.lambda$getBiomes$6(BCLibEndBiomeSource.java:210) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.api.v2.generator.BCLBiomeSource.lambda$getBiomes$4(BCLBiomeSource.java:111) ~[bclib-2.0.4.jar:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[?:?]
	at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[?:?]
	at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[?:?]
	at org.betterx.bclib.api.v2.generator.BCLBiomeSource.getBiomes(BCLBiomeSource.java:113) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.api.v2.generator.BCLibEndBiomeSource.getBiomes(BCLibEndBiomeSource.java:205) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.api.v2.generator.BCLibEndBiomeSource.<init>(BCLibEndBiomeSource.java:82) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.api.v2.generator.BCLibEndBiomeSource.<init>(BCLibEndBiomeSource.java:78) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.api.v2.levelgen.LevelGenUtil.getBCLEndLevelStem(LevelGenUtil.java:80) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.presets.worldgen.BCLWorldPresetSettings.createEndStem(BCLWorldPresetSettings.java:107) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.presets.worldgen.BCLWorldPresetSettings.buildDimensionMap(BCLWorldPresetSettings.java:85) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.presets.worldgen.BCLWorldPresetSettings.buildPreset(BCLWorldPresetSettings.java:71) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.presets.worldgen.BCLWorldPresets.lambda$static$0(BCLWorldPresets.java:32) ~[bclib-2.0.4.jar:?]
	at org.betterx.bclib.presets.worldgen.BCLWorldPresets.bootstrapPresets(BCLWorldPresets.java:101) ~[bclib-2.0.4.jar:?]
	at net.minecraft.class_5317$class_7146.modify$zbk000$bcl_getOverworldStem(class_5317.java:571) ~[client-intermediary.jar:?]
	at net.minecraft.class_5317$class_7146.method_41600(class_5317.java:90) ~[client-intermediary.jar:?]
	at net.minecraft.class_5317.method_41593(class_5317.java:107) ~[client-intermediary.jar:?]
	at net.minecraft.class_5458.method_44104(class_5458.java:98) ~[client-intermediary.jar:?]
	at net.minecraft.class_5458.method_30566(class_5458.java:105) ~[client-intermediary.jar:?]
	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:721) ~[?:?]
	at net.minecraft.class_5458.<clinit>(class_5458.java:104) ~[client-intermediary.jar:?]
	at net.minecraft.class_2378.<clinit>(class_2378.java:326) ~[client-intermediary.jar:?]
	at net.minecraft.class_2966.method_12851(class_2966.java:50) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:154) ~[fabric-loader-0.14.7-1.19.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.7.jar:?]
	... 2 more
```

**Comment by quiqueck** [comment]
Thanks for reporting. This is sort of a duplicate of #2

**Comment by quiqueck** [comment]
Fixed in BCLib 2.0.5.
